src/main/com/aca/exam/exam2/CrudRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.aca.exam.exam2;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+
+public interface CrudRepository<ENTITY, ID> {
+
+    @Nullable
+
+    ENTITY findById(ID id) throws FileNotFoundException;
```

Comment:
The method belongs to an interface and you might have implementations that are not aware of file and hence FileNotFoundException. There was no need to propagate the implementation-specific exception to interface